xpra/server/window/window_video_source.py

```diff
@@ -1079,22 +1087,35 @@ def update_encoding_options(self, force_reload=False):
         """
         super().update_encoding_options(force_reload)
         vs = self.video_subregion
+        log("update_encoding_options(%s): vs: %s, check %s", force_reload, vs, (True if vs else False))
         if vs:
-            if (self.encoding not in ("auto", "grayscale") and self.encoding not in self.common_video_encodings) or \
+            if not (WINDOW_FOCUS_VIDEO_ENCODER and self.has_focus) and \
+                ((self.encoding not in ("auto", "grayscale") and self.encoding not in self.common_video_encodings) or \
                 self.full_frames_only or STRICT_MODE or not self.non_video_encodings or not self.common_video_encodings or \
                 self.content_type=="text" or \
-                self._mmap_size>0:
+                self._mmap_size>0):
                 #cannot use video subregions
                 #FIXME: small race if a refresh timer is due when we change encoding - meh
+                log("cannot use video subregions, reset")
                 vs.reset()
             else:
                 old = vs.rectangle
                 ww, wh = self.window_dimensions
-                vs.identify_video_subregion(ww, wh,
+                
+
+                log("update_encoding_options check setregion override")
+                if WINDOW_FOCUS_VIDEO_ENCODER and self.has_focus:
+                    vs.set_detection(False)
+                    vs.set_region(0, 0, ww, wh)
```

Comment:
So this also bypasses video region detection.
Why was this needed?
Now the whole window is using nvenc, no matter what?

Reply:
yes - the intent was to let the nvenc encoding/compression optimize sending the whole window. I noticed in a lot of cases since partial window refreshes were being sent, the window refreshed in blocks and not all at the same time - since we already are going to try to use nvenc to encode the frame (since the window is in focus) might as well let it handle the optimization and get consistent repaints on the client side.
If there's a better way to do/handle this let me know and I can update